CesiumAsync/src/CachingAssetAccessor.cpp

```diff
@@ -343,20 +346,24 @@ std::string calculateCacheKey(const IAssetRequest& request) {
 std::time_t calculateExpiryTime(
     const IAssetRequest& request,
     const std::optional<ResponseCacheControl>& cacheControl) {
-  if (cacheControl) {
-    if (cacheControl->maxAge() != 0) {
-      return std::time(nullptr) + cacheControl->maxAge();
-    }
-  }
 
-  const IAssetResponse* pResponse = request.response();
-  const HttpHeaders& responseHeaders = pResponse->headers();
-  HttpHeaders::const_iterator expiresHeader = responseHeaders.find("Expires");
-  if (expiresHeader != responseHeaders.end()) {
-    return convertHttpDateToTime(expiresHeader->second);
-  }
+  // If there is a Cache-Control header with the max-age or s-maxage directive
+  // in the response, the Expires header is ignored
+  bool preferCacheControl =
+      cacheControl &&
+      (cacheControl->maxAgeExists() || cacheControl->sharedMaxAgeExists());
+
+  if (preferCacheControl) {
+    return std::time(nullptr) + cacheControl->maxAgeValue();
```

Comment:
Technically we should subtract the value of the `Age` header from the max-age, before adding it to the current time. There's a note about this on the [Mozilla cache-control page](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Cache-Control):

> Note that max-age is not the elapsed time since the response was received; it is the elapsed time since the response was generated on the origin server. So if the other cache(s) — on the network route taken by the response — store the response for 100 seconds (indicated using the Age response header field), the browser cache would deduct 100 seconds from its [freshness lifetime](https://developer.mozilla.org/en-US/docs/Web/HTTP/Caching#freshness_lifetime).

Probably not urgent, but would you mind writing an issue for it?

Reply:
Done. #694